### PR TITLE
Working version

### PR DESCRIPTION
**Pull request by bodik4real**
fixed bugs related to edit and delete functionality.

### CODE REVIEW

File: src/app/login/login/login.component.html [hunk 1]
```diff
@@ -1,15 +1,22 @@
...
+<link href="//netdna.bootstrapcdn.com/bootstrap/3.0.0/css/bootstrap.min.css" rel="stylesheet" id="bootstrap-css">
+<script src="//netdna.bootstrapcdn.com/bootstrap/3.0.0/js/bootstrap.min.js"></script>
+<script src="//code.jquery.com/jquery-1.11.1.min.js"></script>
+
+<div class="container input-group">
+  <div class="row">
+    <div class="col-sm-6 col-md-4 col-md-offset-4">
+      <h1 class="text-center login-title">Sign in to Your task list</h1>
+      <div class="account-wall">
+        <img class="profile-img" src="https://lh5.googleusercontent.com/-b0-k99FZlyE/AAAAAAAAAAI/AAAAAAAAAAA/eu7opA4byxI/photo.jpg?sz=120"
```
Comment (timonsk) [inline src/app/login/login/login.component.html]: Please load an image to img folder and replace it for this link

File: src/app/task-list/task-list-item/task-list-item.component.ts [hunk 2]
```diff
@@ -1,5 +1,6 @@
-import { Component, OnInit, Input, Output, EventEmitter } from '@angular/core';
+import { Component, OnInit, Input, Output, EventEmitter, TemplateRef } from '@angular/core';
 import { TaskItem } from '../../models/TaskItem';
+import { BsModalRef, BsModalService } from '../../../../node_modules/ngx-bootstrap';
```
Comment (timonsk) [inline src/app/task-list/task-list-item/task-list-item.component.ts]: Please use it like  from 'ngx-bootstrap'; Avoid using import like this '../../../../node_modules/

File: src/app/task-list/task-list-item/task-list-item.component.ts [hunk 3]
```diff
@@ -1,5 +1,6 @@
-import { Component, OnInit, Input, Output, EventEmitter } from '@angular/core';
+import { Component, OnInit, Input, Output, EventEmitter, TemplateRef } from '@angular/core';
```
Comment (timonsk) [inline src/app/task-list/task-list-item/task-list-item.component.ts]: Do we really need this TemplateRef  import ?

File: src/app/task-list/task-list-item/task-list-item.component.ts [hunk 4]
```diff
@@ -9,24 +10,37 @@ import { TaskItem } from '../../models/TaskItem';
...
   public isTaskEditing: boolean;
+  modalRef: BsModalRef;
 
   @Input() task: TaskItem;
-  @Output() onTaskEdit = new EventEmitter<TaskItem>();
-  @Output() onTaskDelete = new EventEmitter<number>();
+  @Output() taskEdit = new EventEmitter<TaskItem>();
+  @Output() taskDelete = new EventEmitter<number>();
+  @Output() taskOpenModel = new EventEmitter<any>();
+  constructor( private modalService: BsModalService){
```
Comment (timonsk) [inline src/app/task-list/task-list-item/task-list-item.component.ts]: Add space between constructor and last output

File: src/app/task-list/task-list-item/task-list-item.component.ts [hunk 5]
```diff
@@ -9,24 +10,37 @@ import { TaskItem } from '../../models/TaskItem';
...
+  }
+
+  public startEditing() {
+    this.isTaskEditing = !this.isTaskEditing;
   }
 
-  public startEditing()
-  { 
-    this.isTaskEditing= !this.isTaskEditing;     
+  decline(): void {
```
Comment (timonsk) [inline src/app/task-list/task-list-item/task-list-item.component.ts]: add access modifier  like public or private

File: src/app/task-list/task-list.component.ts [hunk 6]
```diff
@@ -16,6 +16,7 @@ export class TaskListComponent {
 
   public tasks = new Array<TaskItem>();
   public IsHidden = true;
+  modalRef: BsModalRef;
```
Comment (timonsk) [inline src/app/task-list/task-list.component.ts:19]: Add access modifier

File: src/app/task-list/task-list.module.ts [hunk 7]
```diff
@@ -19,7 +19,10 @@ import { TaskListItemComponent } from './task-list-item/task-list-item.component
     TaskListComponent,
     TaskListItemComponent
   ],
-  exports: [TaskListComponent],
+  exports: [
+    TaskListComponent,
```
Comment (timonsk) [inline src/app/task-list/task-list.module.ts:23]: Format

File: src/app/login/login/login.component.html (hunk 1 above)
Comment (bodik4real) [reply src/app/login/login/login.component.html]: Image is replaced

File: src/app/task-list/task-list-item/task-list-item.component.ts (hunk 2 above)
Comment (bodik4real) [reply src/app/task-list/task-list-item/task-list-item.component.ts]: Used

File: src/app/task-list/task-list-item/task-list-item.component.ts (hunk 3 above)
Comment (bodik4real) [reply src/app/task-list/task-list-item/task-list-item.component.ts]: You’re right. We needed TemplateRef for open Modal method. But now this method is also useless, So, I removed both of them.

File: src/app/task-list/task-list-item/task-list-item.component.ts (hunk 4 above)
Comment (bodik4real) [reply src/app/task-list/task-list-item/task-list-item.component.ts]: done

File: src/app/task-list/task-list-item/task-list-item.component.ts (hunk 5 above)
Comment (bodik4real) [reply src/app/task-list/task-list-item/task-list-item.component.ts]: added

File: src/app/task-list/task-list.component.ts (hunk 6 above)
Comment (bodik4real) [reply src/app/task-list/task-list.component.ts:19]: added

File: src/app/task-list/task-list.module.ts (hunk 7 above)
Comment (bodik4real) [reply src/app/task-list/task-list.module.ts:23]: Fixed

File: src/app/login/login/login.component.html [hunk 8]
```diff
@@ -1,15 +1,22 @@
...
+<link href="//netdna.bootstrapcdn.com/bootstrap/3.0.0/css/bootstrap.min.css" rel="stylesheet" id="bootstrap-css">
+<script src="//netdna.bootstrapcdn.com/bootstrap/3.0.0/js/bootstrap.min.js"></script>
+<script src="//code.jquery.com/jquery-1.11.1.min.js"></script>
+
+<div class="container input-group">
+  <div class="row">
+    <div class="col-sm-6 col-md-4 col-md-offset-4">
+      <h1 class="text-center login-title">Sign in to Your task list</h1>
+      <div class="account-wall">
+        <img class="profile-img" src="./../../../assets/Images/photo.png"
```
Comment (timonsk) [inline src/app/login/login/login.component.html]: Please rename folder to lowercase

**Pull request merged**